micro-rdk/src/common/conn/viam.rs

```diff
@@ -707,24 +738,33 @@ where
     }
 
     async fn connect_to_app(&self) -> Result<AppClient, AppClientError> {
-        let robot_creds = self.storage.get_robot_credentials().unwrap();
         let app_uri = self
             .storage
             .get_app_address()
             .unwrap_or("https://app.viam.com:443".parse::<Uri>().unwrap());
-        let app_client_io = self
-            .http2_connector
-            .connect_to(&app_uri)
-            .map_err(AppClientError::AppClientIoError)?
-            .await
-            .map_err(AppClientError::AppClientIoError)?;
-        let grpc_client = GrpcClient::new(app_client_io, self.executor.clone(), app_uri)
-            .await
-            .map_err(AppClientError::AppGrpcClientError)?;
 
-        AppClientBuilder::new(Box::new(grpc_client), robot_creds.clone())
-            .build()
-            .await
+        let robot_creds = self.storage.get_robot_credentials().unwrap();
+
+        {
```

Comment:
Moved the statements with `?` into a block so I can `inspect` any failure before returning, but otherwise this code didn't change except for logs.